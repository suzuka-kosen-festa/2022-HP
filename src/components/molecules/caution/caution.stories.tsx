import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Caution } from ".";

type T = typeof Caution
type Story = ComponentStoryObj<T>

const data = {
  title: "バザーに関するお願い",
  sentence: [
      "学内における飲食物の食べ歩きはご遠慮ください。",
      "学内バザーにて購入した飲食物のお持ち帰りはご遠慮ください",
      "各出店バザーにお並びいただく際、1つの出店につきお並びいただく人数は5人までとしておりますので、当日は係員の指示に従ってください。"
  ]
}

export default {
  args: {
    title: data.title,
    sentence: data.sentence
  },
  argTypes: {
    title: {
      description: "Title of caution",
      control: {
        type: "text"
      }
    },
    sentence: {
      description: "Detailed precautions",
      control: {
        type: "array"
      }
    }
  },
  component: Caution
} as ComponentMeta<T>

export const Default: Story = {}
