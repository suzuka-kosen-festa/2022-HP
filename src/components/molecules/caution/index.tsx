import tw from "twin.macro";
import type { FC } from "react";
import { CautionProperties } from "./types/model";
import Icon from "../../atoms/icon";
import { SubHeading } from "../../atoms/sub-heading";

const CautionBox = tw.section`flex flex-col justify-center items-center border-2 pt-2 pb-4 px-4.5 sm:(border-4 pt-4 pb-8 px-9) border-caution`;

const CautionTitle = tw.div`flex space-x-1 mb-2 sm:(space-x-2 mb-4)`;

const Caution: FC<CautionProperties> = ({ title, sentence, ...rest }) => (
  <CautionBox aria-label="注意事項" role="alert" {...rest}>
    <CautionTitle>
      <Icon icon="ant-design:warning-filled" iconStyles={tw`text-caution`} />
      <SubHeading css={tw`flex self-center`}>{title}</SubHeading>
    </CautionTitle>
    <ul css={tw`list-disc list-outside space-y-0.5 sm:space-y-2`}>
      {sentence.map(value => (
        <li css={tw`ml-4.5`} key={`${value}`}>
          {value}
        </li>
      ))}
    </ul>
  </CautionBox>
);

export { Caution };
