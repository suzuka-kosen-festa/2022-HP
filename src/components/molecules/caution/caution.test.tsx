import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./caution.stories";

const { Default } = composeStories(stories);

const options = {
  name: "注意事項"
}

describe("(components) molecules/twitter", () => {
  test("to be molecules", () => {
    const { container } = render(<Default />);
    expect(container).toBeMolecule();
  });
  test("to be [role=alert]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("alert", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
