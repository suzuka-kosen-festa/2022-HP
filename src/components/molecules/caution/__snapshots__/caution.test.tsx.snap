// Vitest Snapshot v1

exports[`(components) molecules/twitter > take snap shot 1`] = `
<div>
  <section
    aria-label="注意事項"
    class="css-u8u4n6"
    role="alert"
  >
    <div
      class="css-1m7spai"
    >
      <div
        class="css-cw7xep"
      >
        <span />
      </div>
      <h2
        class="css-qt9rcz-Caution"
      >
        バザーに関するお願い
      </h2>
    </div>
    <ul
      class="css-mvofyw-Caution"
    >
      <li
        class="css-u80qbw-Caution"
      >
        学内における飲食物の食べ歩きはご遠慮ください。
      </li>
      <li
        class="css-u80qbw-Caution"
      >
        学内バザーにて購入した飲食物のお持ち帰りはご遠慮ください
      </li>
      <li
        class="css-u80qbw-Caution"
      >
        各出店バザーにお並びいただく際、1つの出店につきお並びいただく人数は5人までとしておりますので、当日は係員の指示に従ってください。
      </li>
    </ul>
  </section>
</div>
`;
